packages/trader/src/AppV2/Components/TradeParameters/Multiplier/multiplier-wheel-picker.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useEffect } from 'react';
+import { ActionSheet, Text, WheelPicker, Skeleton } from '@deriv-com/quill-ui';
+import { Localize } from '@deriv/translations';
+import { useTraderStore } from 'Stores/useTraderStores';
+import debounce from 'lodash.debounce';
+
+type TMultiplierWheelPickerProps = {
+    multiplier: ReturnType<typeof useTraderStore>['multiplier'];
+    multiplier_range_list: ReturnType<typeof useTraderStore>['multiplier_range_list'];
+    currency: ReturnType<typeof useTraderStore>['currency'];
+    commission: ReturnType<typeof useTraderStore>['commission'];
+    setMultiplier: (multiplier: number) => void;
+};
+
+const MultiplierWheelPicker = ({
+    multiplier,
+    multiplier_range_list = [],
+    currency,
+    commission,
+    setMultiplier,
+}: TMultiplierWheelPickerProps) => {
+    const multiplier_array = multiplier_range_list.map(item => ({ value: item.text }));
+    const initial_multiplier = React.useRef<number>();
+    const selected_multiplier = React.useRef<number>(multiplier);
+
+    const debouncedSetMultiplier = debounce((setMultiplier, multiplier) => {
```

Comment:
`debouncedSetMultiplier` should be moved outside of the component if you want to use `cancel` method

Reply:
cancel doesnt come first and the component unmount finishes ?